### ISSUE TITLE
Update dict values only with some patterns

### ISSUE THREAD

**Issue by williamjoy** [mikefarah/yq#1201]
**Please describe your feature request.**
With an array or dict, support update some of the elements based on dynamic conditions, and keep rest on changed
Eg. I wish I could use yq to  convert k8s configmap data from raw string to yaml, if the data key ends with `.yaml` `.yml` `json` etc

The purpose is to use  git`textcov` feature to generate better diffs for k8s configmaps
https://git.wiki.kernel.org/index.php/Textconv

Git supports to define file patterns and pass it to external command before doing diffs, in this case I'm trying to convert `**/configmap.yaml` using `yq` to convert raw strings to yaml

Note:
- how to questions should be posted in the discussion board and not raised as an issue.
- V3 will no longer have any enhancements.

**Describe the solution you'd like**
If we have data1.yml like:

```yaml
---
# Source: chart/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: app-default-9d079a8346
data:
  config.yaml: |-
    test: 1
  liveness.sh: "ping -c 1 xxx"
  check.py: |-
    python code
  something.ini: |-
    [section]
  app_conf.yaml: |-
    test: 1
  db_conf.json: |-
    {"db": "localhost", xxx}

```

And we run a command: 

```bash
yq '.data = (.data.[] |= (.|from_yaml))'

# possibly to support this?
with_etries( .key | test('.*\.yaml') ? (.value | from_yaml) : .value)
```

it could output

```yaml
# Source: chart/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: app-default-9d079a8346
data:
  config.yaml: 
    test: 1
  liveness.sh: "ping -c 1 xxx"
  check.py: |-
    python code
  something.ini: |-
    [section]
  app_conf.yaml: 
    test: 1
  db_conf.json:
    db: localhost
    xxx: 3306
```

**Describe alternatives you've considered**

**Additional context**

**Comment by mikefarah** [comment]
TLDR:

```
yq '(.data[] |  select(key | test("\.(yaml|json|yml)$")) ) |= (from_yaml | ... style = "")' file.yaml
```

Basically this:
- splats the data entries
- filters to only includes keys ending with json/yaml/yml
- updates those matches via from_yaml and removes any styling (so that json is formatted as idiomatic yaml).

**Comment by williamjoy** [comment]
Thanks! This works perfectly! 💯 
(I wanted to open a `question` type issue but there was no so such choice, so I chose feature request instead)

**Comment by mikefarah** [comment]
Ah ok - use the 'discussion' section for questions https://github.com/mikefarah/yq/discussions

**Comment by williamjoy** [comment]
Oh, I haven't used this GitHub feature before, definitely will look into more ideas there.

I've figured out how to escape quotes to make it working with git. ( use `yq -v` to show what arguments yq gets)

Not sure what is best place to share it,  I'll post it in discussion hopefully that's proper place to share.

```bash
# ~/.gitconfig
[diff "configmap"]
    textconv = yq '(.data[]|select(key|test(\"\\.(yml|yaml|json)$\")) ) |= (from_yaml | ... style = \"\")'
```

```bash
# ~/.gitattributes

**/manifests/**/*configmap.yaml diff=configmap
**/manifests/**/*config-map.yaml diff=configmap
```

Run git command
```bash
git  log -p --textconv  '**/*configmap.yaml'
```
